### PR TITLE
Diane.kim addpullsecret

### PR DESCRIPTION
**Pull request by dianekim1995**
add_image_pullsecret logic wasn't actually adding to an existing list, but overwriting existing pull secrets.
This a fix to that logic.

### CODE REVIEW

File: operator_csv_libs/csv.py [hunk 1]
```diff
@@ -108,18 +108,24 @@ def add_image_pullsecret(self, name):
...
-            d['spec']['template']['spec']['imagePullSecrets'] = p    
+                # If imagepullsecret is missing, set it
+                d['spec']['template']['spec']['imagePullSecrets'] = p
+            else:
+                # If imagepullsecret exists, add to the list
+                if not(all(x in p for x in d['spec']['template']['spec']['imagePullSecrets'])):
+                    if isinstance(d['spec']['template']['spec']['imagePullSecrets'], list):
+                        d['spec']['template']['spec']['imagePullSecrets'].extend(p)
+                    else:
+                        raise Exception('imagePullSecrets is not of type list')
```
Comment (hassenius) [inline operator_csv_libs/csv.py:128]: Might be better to add the `TypeError` exception, then a generic `Exception`?

**Comment by hassenius** [comment]
Thanks for the PR @dianekim1995 
Seems like there's an issue with the unit test now: https://github.com/multicloud-ops/operator-csv-libs/pull/38/checks?check_run_id=1864641799
Not sure if the original test case was incorrect, or if the code change breaks some existing (correct) behaviour?

File: operator_csv_libs/csv.py (hunk 1 above)
Comment (dianekim1995) [reply operator_csv_libs/csv.py:128]: Will change that!

**Comment by dianekim1995** [comment]
@hassenius Thanks for catching that. I updated the test logic as well.

**Review by eveliososa — APPROVED** [review]
LGTM! I ran the test on the previous code and the new code and it's not overwriting the pull secrets any more. Thanks Diane for catching this!
Screen Shot 2021-02-10 at 8 47 13 AM
Screen Shot 2021-02-10 at 8 38 58 AM

**Review by hassenius — APPROVED** [review]
LGTM

**Pull request merged**